### PR TITLE
Check cStringUsingEncoding for failure and return nil to indicate failure. Fixes #211.

### LINKED ISSUE #211: Crash with base64DecodeString
I came across a crash in BNCEncodingUtils in base64DecodeString where objPointer was NULL.
``` 
const char * objPointer = [strBase64 cStringUsingEncoding:NSASCIIStringEncoding];
intLength = strlen(objPointer);
```
This happened a couple times in a row but I was unable to reproduce it afterwards (sorry!) so I couldn't get a backtrace. strlen does not handle null and consequently causes a crash.

### PR DESCRIPTION
**Pull request by drichardson**

### CODE REVIEW

**Comment by drichardson** [comment]
Just wanted to note that before I made any changes 7 unit tests were failing in the branch SDK when built with Xcode 7.1 (7B91b) running under iPhone 6 simulator. Looks like those same tests failed during the circleci integration.

File: Branch-SDK/Branch-SDK/BNCEncodingUtils.m [hunk 1]
```diff
@@ -41,7 +41,11 @@ + (NSString *)base64EncodeStringToString:(NSString *)strData {
 }
 
 + (NSString *)base64DecodeStringToString:(NSString *)strData {
-    return [[NSString alloc] initWithData:[BNCEncodingUtils base64DecodeString:strData] encoding:NSUTF8StringEncoding];
+    NSData* data =[BNCEncodingUtils base64DecodeString:strData];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCEncodingUtils.m]: This doesn't quite match coding standards, prefer NSData *data = [...

File: Branch-TestBed/Branch-SDK-Tests/BNCEncodingUtilsTests.m [hunk 2]
```diff
@@ -216,6 +216,12 @@ - (void)testDecodeBase64EncodedJsonStringToDictionary {
     XCTAssertEqualObjects(decodedValue, expectedDataDict);
 }
 
+- (void)testDecodeNonASCIIString {
+    // Should fail, but not crash.
+    NSString* result = [BNCEncodingUtils base64DecodeStringToString:@"𝄞"];
+    XCTAssertNil(result);
```
Comment (shortstuffsushi) [inline Branch-TestBed/Branch-SDK-Tests/BNCEncodingUtilsTests.m]: Same formatting note.

**Comment by drichardson** [comment]
What coding standards? I didn't see any in the readme.

**Comment by shortstuffsushi** [comment]
They aren't written out, just try to match existing code.

**Comment by drichardson** [comment]
OK, well you can take this or leave it. Not going to make another fix for it.

**Comment by shortstuffsushi** [comment]
Will probably pull in the branch and make the changes before merging. Mostly a note to self. Thank you for the contribution, though, we do really appreciate it.

**Comment by derrickstaten** [comment]
@drichardson thanks so much for your contribution. We're going to review this today.

**Comment by derrickstaten** [comment]
:+1:

**Pull request merged**